source/extensions/tracers/opentelemetry/http_trace_exporter.cc

```diff
@@ -0,0 +1,79 @@
+#include "http_trace_exporter.h"
+
+#include <chrono>
+#include <memory>
+#include <string>
+
+#include "source/common/common/logger.h"
+#include "source/common/protobuf/protobuf.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace OpenTelemetry {
+
+OpenTelemetryHttpTraceExporter::OpenTelemetryHttpTraceExporter(
+    Upstream::ClusterManager& cluster_manager, envoy::config::core::v3::HttpService http_service,
+    OpenTelemetryTracerStats& tracing_stats)
+    : cluster_manager_(cluster_manager), http_service_(http_service),
+      tracing_stats_(tracing_stats) {}
+
+bool OpenTelemetryHttpTraceExporter::log(const ExportTraceServiceRequest& request) {
+
+  std::string request_body;
+
+  const auto ok = request.SerializeToString(&request_body);
+  if (!ok) {
+    ENVOY_LOG(warn, "Error while serializing the binary proto ExportTraceServiceRequest.");
+    return false;
+  }
+
+  const auto thread_local_cluster =
```

Comment:
```suggestion
  const auto* thread_local_cluster =
```

Reply:
That doesn't seem to work:

```
'this' argument to member function 'httpAsyncClient' has type 'const ThreadLocalCluster', but function is not marked constclang(member_function_call_bad_cvr)
thread_local_cluster.h(144, 30): 'httpAsyncClient' declared here
```

Other usages are also like that.